md-to-pl.py

```diff
@@ -0,0 +1,109 @@
+import os
+import uuid
+import json
+import md_parser
+
+# TODO: validity checking for customizations
+ALLOWED_CUSTOMIZATIONS = {'multiple-choice': ['answers-name', 'weight', 'inline', 'number-answers', 'fixed-order',
+                                              'hide-letter-keys', 'all-of-the-above', 'none-of-the-above',
+                                              'external-json', 'external-json-correct-key',
+                                              'external-json-incorrect-key'],
+                          'checkbox': ['answers-name', 'weight', 'inline', 'number-answers', 'min-correct',
+                                       'max-correct', 'fixed-order', 'partial-credit', 'partial-credit-method',
+                                       'hide-help-text', 'detailed-help-text', 'hide-answer-panel', 'hide-letter-keys',
+                                       'hide-score-badge'],
+                          'number-input': ['answers-name', 'weight', 'correct-answer', 'label', 'suffix', 'display',
+                                           'comparison', 'rtol', 'atol', 'digits', 'allow-complex', 'allow-blank',
+                                           'blank-value', 'show-help-text', 'show-placeholder', 'size',
+                                           'show-correct-answer', 'allow-fractions']}
+
+# TODO: add rest of types
+SINGLE = ['number-input']
+MULTIPLE = ['multiple-choice', 'checkbox']
+
+# Create input and output file directories if they DNE
+try:
+    os.mkdir("question_templates")
+    input("> Place MD question files in 'question_templates' folder.\n"
+          "> Press Enter to continue\n")
+except FileExistsError:
+    pass
+try:
+    os.mkdir("PL-questions")
+except FileExistsError:
+    pass
+
+# Check for question inputs
+_, _, filenames = next(os.walk("question_templates"))
+if not filenames:
+    input("> Question source folder empty\n"
+          "> Press Enter to quit\n")
+    quit()
+
+# Process questions
+for q_name in filenames:
+    parsed_q = md_parser.parse("question_templates/" + q_name)
+
+    # Create output dir
+    out_dir = 'PL-questions/' + parsed_q['header']['title']
+    try:
+        os.mkdir(out_dir)
+    except FileExistsError:
+        pass
+
+    # Writing to files
+    # Creating info.json file
+    with open(out_dir + "/info.json", 'w') as info:

```

Comment:
Use `pathlib` for a newer more pythonic version of dealing with files

[Source](https://realpython.com/python-pathlib/)

Reply:
resolved in #20